design/sd-local.md

```diff
@@ -0,0 +1,164 @@
+- SD_ARTIFACTS_DIR: `/sd/workspace/artifacts`
+- SD_META_PATH: `/sd/meta/meta.json`
+- SD_ROOT_DIR: `/sd/workspace`
+- SD_SOURCE_DIR: `/sd/workspace/src/<SCM hostname>/<organization>/<repository>`
+- SD_SOURCE_PATH: Generate from `screwdriver.yaml`
+- SD_CONFIG_DIR: Not set (Need to be set when External Config is supported)
+- CONFIG_URL: Not set (Need to be set when External Config is supported)
+- SD_API_URL: The value got from config file
+- SD_STORE_URL: The value got from config file
+- CI: `false`
+- CONTINUOUS_INTEGRATION: `false`
+- SCREWDRIVER: `false`
+
+
+## Usage
+
+### Prerequisites
+- Docker runtime
+
+### Start build
+
+```bash
+$ sdlocal build [job-name] [options]
+```
+
+#### Input
+
+- `job-name` the job in `screwdriver.yaml` to be run locally (default: `./screwdriver.yaml`)
+
+##### Options
+
+- `--meta [json]` Set values of `meta` (JSON)
+- `--meta-file [path]` Path to config file of `meta` (JSON)
+- `-e, --env [key=value]` Set `key` and `value` relationship which is set as environment variables of Build Container.
+  - `secrets` is also set as environment variables.
+- `--env-file [path]` Path to config file of environment variables. (`.env`)
+- `--artifacts-dir [path]` Path to the host side directory which is mounted into `$SD_ARTIFACTS_DIR`. (default: `./sd-artifacts`)
+- `-m, --memory [size]` Set memory size which Build Container can use. Either b, k, m, g can be used as a size unit. (default: ?)
+- `-o --output [path]` Path to file name where the logs are output. (default: stdout)
+- `--src-url [repository url]` Set repository URL which is to build when user use the remote repository without local files.
+
+#### Output
+
+- Logs of all executed steps in the job.
+- Artifacts which is output to `$SD_ARTIFACTS_DIR`.
+
+### Configuration
+
+```bash
+$ sdlocal config set [key] [value]
+```
+
+The chart below shows relationship between `key` and `value`.
+
+|key|value|
+|:-:|:-:|
+|api-url|Screwdriver API URL|
+|store-url|Screwdriver Store URL|
+|token|Screwdriver API token|
+|launcher-version|Version of Launcher Image (default: stable)|
+|launcher-image|Name of Launcher Image (default: screwdrivercd/launcher)|
+
+Create config file as `$HOME/.sdlocal/config` with YAML format.
+
+```yaml
+api-url: <Screwdriver API URL>
+store-url: <Screwdriver Store URL>
+token: <Screwdriver API Token>
+launcher:
+  version: <Launcher Version>
+  image: <Launcher image name>
+```
+
+Can confirm setting configurations with the command below:
+
+```bash
+$ sdlocal config view
+```
+
+#### Options
+
+- `--local` Run command with `.sdlocal/config` file in current directory.
+
+## Design considerations
+
+#### Implement Local Mode Launcher.
+Need to be implement the following:
+- Add `--local-mode` option to run Launcher binary on Local Mode
+- Not to call Screwdriver API on Local Mode
+- Use the steps information from the response of validator API in `Initialization`
+- Output the build logs to stdout instead of `log-service`
```

Comment:
one option is to support a `local-mode` in `log-service` and have it write to `console`. Also we can leverage log-service to write step logs under a directory in `artifacts` folder, then `sdlocal` can simply `tail` this file.

Reply:
Is it easier to use `log-service`  to write step logs under a directory in `artifacts` folder?
Do you think you don't need to output logs to stdout?

Reply:
It's sounds good. It allows sdlocal to decorate logs. For example, it can put the step name between steps.

Reply:
Fixed to use `log-service` and output logs to Artifacts Directory.